generators/service-watson-conversation/templates/node/instrumentation.js

```diff
@@ -1,12 +1,28 @@
 const IBMCloudEnv = require('ibm-cloud-env');
-const ConversationV1 = require('watson-developer-cloud/conversation/v1');
+const AssistantV1 = require('watson-developer-cloud/assistant/v1');
 
 module.exports = function(app, serviceManager){
-	const conversation = new ConversationV1({
+	let params = {
+		version: '2018-02-16',
 		url: IBMCloudEnv.getString('watson_conversation_url'),
-		username: IBMCloudEnv.getString('watson_conversation_username'),
-		password: IBMCloudEnv.getString('watson_conversation_password'),
-		version_date: ConversationV1.VERSION_DATE_2017_04_21
-	});
-	serviceManager.set("watson-conversation", conversation);
+	};
+
+	if (IBMCloudEnv.getString('watson_conversation_apikey')) {
+		const iam_url = params.url.includes('gateway-s.') ? 
+			'https://iam.stag1.ng.bluemix.net/identity/token' :
+			'https://iam.ng.bluemix.net/identity/token';
```

Comment:
The urls support only `us-south` region?

Reply:
stag1?

Reply:
I agree having internal IBM systems exposed in a public git repo is not good, I would remove that piece

Reply:
good catch @christiancompton.

@mjperrins we can hide `iam url` and keep it optional and SDKs can handle that. But the problem is that for stage1 it will not work because default iam url used in SDKs is `https://iam.bluemix.net/identity/token`. Is there a way to get the `iam_url` from env variable?